smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/integration/serde/json/MemberShapeDecodeGenerator.kt

```diff
@@ -63,9 +64,14 @@ abstract class MemberShapeDecodeGenerator(
         if (member.hasTrait(SwiftBoxTrait::class.java)) {
             symbol = symbol.recursiveSymbol()
         }
-        val decodeVerb = if (symbol.isBoxed()) "decodeIfPresent" else "decode"
+        val defaultValue = symbol.defaultValue()
+        val decodeVerb = if (symbol.isBoxed() || !defaultValue.isNullOrEmpty()) "decodeIfPresent" else "decode"
         val decodedMemberName = "${memberName}Decoded"
-        writer.write("let \$L = try \$L.$decodeVerb(\$N.self, forKey: .\$L)", decodedMemberName, containerName, symbol, memberName)
+
+        // no need to assign nil to a member that is optional
+        val defaultValueLiteral = if (defaultValue != null && defaultValue != "nil") "?? $defaultValue" else ""
+
+        writer.write("let \$L = try \$L.$decodeVerb(\$N.self, forKey: .\$L) $defaultValueLiteral", decodedMemberName, containerName, symbol, memberName)
```

Comment:
nit: when there is no default value, this will result in trailing space.  Do we care about such things?